src/main.py

```diff
@@ -10,70 +10,42 @@
 
 Authors: Zein Youssef and Steven Dropiewski
 """
-from time import sleep
-import os
-
-from bluetoothctl import Bluetoothctl
 from follower import Follower
 from leader import Leader
 from tagrec import TagRecognition
 
 
-def is_controller_connected():
-    """
-    Determines if a Bluetooth controller is connected to /dev/input/js0.
-
-    :return: True if a device is connected and False if not.
-    :rtype: Boolean
+def decide_role(tag_data):
     """
-    return os.path.exists('/dev/input/js0')
+    Creates a Leader or Follower object.
 
+    :param tag_data: Return value of TagRecognition.detect().
 
-def disconnect_and_remove_device(bt, bt_addr):
+    :return: If tag_data is None, the vehicle will become a Leader. Otherwise,
+             it will become a Follower.
+    :rtype: Leader or Follower
     """
-    Disconnects and removes a controller from Bluetooth.
-
-    :param bt: The object used to send Bluetooth commands.
-    :type bt: Bluetoothctl
+    if tag_data:
+        vehicle = Follower()
+    else:
+        vehicle = Leader()
 
-    :param bt_addr: The Bluetooth address of the device to disconnect from
-                    and remove.
-    :type bt_addr: string
-    """
-    bt.disconnect(bt_addr)
-    sleep(Follower.CYCLE_TIME)
-    if any(d['mac_address'] == bt_addr for d in bt.get_paired_devices()):
-        bt.remove(bt_addr)
+    return vehicle
 
 
 def main():
     """
     The main driver program for the robotic leader-follower software.
     """
-    tag = TagRecognition()
-    bt = Bluetoothctl()
-    BT_ADDR = "5C:BA:37:26:6D:9A"
+    tag = TagRecognition(marker_length=0.025)
+    vehicle = decide_role(tag.detect())
 
-    # If an ARTag is detected, the vehicle will become a follower.
-    # If an ARTag is not detected, the vehicle will become a leader.
-    if tag.detect():
-        follower = Follower()
+    if isinstance(vehicle, Follower):
         while True:
```

Comment:
Could we get rid of the if else statements and just turn them into while isinstance(vehicle, Follower)? Since the vehicles won't switch roles while running. Or make the while true statement and then have the 2 if statements inside of it?